src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -16,222 +16,75 @@
-
-	/**  getter for kernel width
-	 * @return kernel width
-	 * throws exception if kernelwidth <=0
-	 */
-	float64_t get_kernelwidth( ) const;
-
-	/**  getter for the random coefficients
-	 * necessary for creating random fourier features compatible to the current ones
-	 * returns values of internal members randomcoeff_additive
-	 * and randomcoeff_multiplicative
-	 */
-	void get_randomcoefficients(float64_t ** randomcoeff_additive2,
-			float64_t ** randomcoeff_multiplicative2,
-			int32_t *dim_feature_space2, int32_t *dim_input_space2, float64_t* kernelwidth2 ) const;
-
-	/**  setter for the random coefficients
-	 * necessary for creating random fourier features compatible to the previous ones
-	 * sets values of internal members	randomcoeff_additive
-	 * and randomcoeff_multiplicative
-	 * simply use as input what you got from get_random_coefficients(...)
-	 */
-	void set_randomcoefficients(float64_t *randomcoeff_additive2,
-			float64_t * randomcoeff_multiplicative2,
-			const int32_t dim_feature_space2, const int32_t dim_input_space2, const float64_t kernelwidth2);
-
-	/** a setter
-	 * @param dim the value of protected member dim_input_space
-	 * throws a shogun exception if dim<=0
-	 */
-	void set_dim_input_space(const int32_t dim);
-
-	/** a setter
-	 * @param dim the value of protected member dim_feature_space
-	 * throws a shogun exception if dim<=0
-	 *
-	 */
-	void set_dim_feature_space(const int32_t dim);
-
-	/** computes new random coefficients IF test_rfinited() evaluates to false
-	 * test_rfinited() evaluates to TRUE if void set_randomcoefficients(...)
-	 * hase been called and the values set by set_dim_input_space(...) ,
-	 * set_dim_feature_space(...) and set_kernelwidth(...) are consistent to the
-	 * call of void set_randomcoefficients(...)
+	/** @brief Preprocessor that approximates Gaussian feature map.
 	 *
-	 * throws shogun exception if dim_feature_space <= 0 or dim_input_space <= 0
+	 * Gaussian kernel of form K(x,y) = exp(-||x - y||^2 / kernelwidth)
+	 * is approximated using the formula: 
 	 *
-	 * @return returns true if test_rfinited() evaluates to false and new
-	 * coefficients are computed
-	 * returns false if test_rfinited() evaluates to true and old random
-	 * coefficients are kept which were set by a previous call to void
-	 * set_randomcoefficients(...)
-	 *
-	 * this function is useful if you want to use apply_to_feature_vector but
-	 * cannot call before it fit(Features *f)
+	 * \f[
+	 * 	z(x) = \sqrt{2/D}\ cos(w'x + b)
+	 *  K(x, y) = \langle\phi(x),\phi(y)\rangle \approx z'(x)z(y)
+	 * \f]
+	 * 
+	 * Reference:
+	 * [1] Rahimi, A., & Recht, B. (2008). Random features for large-scale kernel
+	 * machines. In Advances in neural information processing systems (pp.
+	 * 1177-1184).
 	 *
 	 */
-	bool init_randomcoefficients();
+class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>> {
+public:
+		/** default constructor */
+		RandomFourierGaussPreproc();
```

Comment:
think your whitespaces are messed up here

Reply:
@karlnapf  it looks good on my local machine :( idk why this is happening

Reply:
it seems like you have two tab spaces instead of one?

Reply:
Hmm, fixed!